statgen/raremetal.js
# Remove "eval" usage from numeric.js

Refs #15

# Purpose
Numeric.js uses `eval`, which is blocked by some CSPs. This means that raremetal.js would not be allowed to load and run in principle. Fortunately, the `eval` usages do not affect the library functionality that we depend on.

This is intended to be the minimum possible set of changes, manually removing one block from the built bundle. Other functions are not altered, even when they also reference ccs features.

We will now ship our own copy of numeric.js inside the repo with these modifications. This is fairly unusual, but the parent library has not been updated since 2012, and even its website is "[finished](https://groups.google.com/g/numericjs/c/4_g1mFihbIE/m/amMAr7-NCQAJ)" (in the words of the author).

# How to test this
* Run `npm install` to update dependencies. Verified all unit tests passing (`npm run test`).
* Verified same calculation results in `example.html` with built assets (`npm run build`).
* Manually inspected code for other functions that indirectly referenced the dynamically created `numeric.ccs*` functions in the `eval` block; none found.
* Inspected calculation results in the locuszoom.js demo, using a local copy of built assets. Saw exact same results with 1 pheno/many masks/all 4 methods.
* Checked documentation for method details in case of side effects (the website is toast; see [wayback machine](https://web.archive.org/web/20180828221105/http://numericjs.com/documentation.html))

## The road not traveled
* Deeper validation per `/net/snowwhite/home/welchr/projects/covarmatrices/README.md` : this may be evaluated during final validation of Alan's conditional analysis branch.
* I attempted to build library from scratch in parent repo with [build script](https://github.com/sloisel/numeric/blob/656fa1254be540f428710738ca9c1539625777f1/tools/build.sh), but even an unmodified version has failing unit tests.  (the library is unmaintained and this [may be a known issue](https://groups.google.com/g/numericjs/c/BqI7_GRX_ZY/m/0lSGKo8qAAAJ))
* Considered replacing numeric.js with another linear algebra library, but we depend on a number of specific methods, and test coverage/quality of other libraries varies widely; changing the library would require revalidating our results at a deeper level.
